.github/workflows/test_blas.yml

```diff
@@ -0,0 +1,95 @@
+name: TheRock BLAS math-lib tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+      amdgpu_families:
+        type: string
+        default: gfx94X-dcgpu
+      runs-on:
+        type: string
+        default: linux-mi300-1gpu-ossci-rocm
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+      amdgpu_families:
+        type: string
+        default: gfx94X-dcgpu
+      runs-on:
+        type: string
+        default: linux-mi300-1gpu-ossci-rocm
+  push:
+    branches:
+      - ADHOCBUILD
+
+jobs:
+  test_blas:
+    name: "BLAS math-lib test"
+    runs-on: ${{ inputs.runs-on }}
+    strategy:
+      fail-fast: false
+    defaults:
+      run:
+        shell: bash
+    env:
+      VENV_DIR: ${{ github.workspace }}/.venv
+      ARTIFACT_RUN_ID: "${{ inputs.artifact_run_id != '' && inputs.artifact_run_id || github.run_id }}"
+      BUILD_ARTIFACTS_DIR: "./build/artifacts"
+      THEROCK_BIN_DIR: "./build/artifacts/output_dir/bin"
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Install the AWS tool
+        run: ./dockerfiles/cpubuilder/install_awscli.sh
+
+      - name: "Setting up Python"
+        id: setup_python
+        uses: actions/setup-python@42375524e23c412d93fb67b49958b491fce71c38 # v5.4.0
+        with:
+          python-version: 3.11
+
+      - name: Create Python venv and installing packages
+        run: |
+          python -m venv ${VENV_DIR}
+          source ${VENV_DIR}/bin/activate
+          pip install -r requirements-test.txt
+          sudo apt install gfortran -y
```

Comment:
I think `libgfortran5` might be enough and you don't need the full compiler but I would suggest to leave as is (if tests pass) and test / rework in a follow up.